src/handlers/tracker.ts

```diff
@@ -0,0 +1,160 @@
+import { IEditorTracker } from '@jupyterlab/fileeditor';
+
+import { INotebookTracker } from '@jupyterlab/notebook';
+
+import { IConsoleTracker } from '@jupyterlab/console';
+
+import { IDisposable } from '@phosphor/disposable';
+
+import { Signal } from '@phosphor/signaling';
+
+import { Callstack } from '../callstack';
+
+import { EditorHandler } from './editor';
+
+import { Debugger } from '../debugger';
+
+import { IDebugger } from '../tokens';
+
+export class TrackerHandler implements IDisposable {
+  constructor(options: DebuggerWidgetHandler.IOptions) {
+    this.debuggerService = options.debuggerService;
+    this.notebookTracker = options.notebookTracker;
+    this.consoleTracker = options.consoleTracker;
+    this.editorTracker = options.editorTracker;
+
+    this.debuggerService.modelChanged.connect(() => {
+      const debuggerModel = this.debuggerService.model as Debugger.Model;
+
+      debuggerModel.callstackModel.currentFrameChanged.connect(
+        this.onCurrentFrameChanged,
+        this
+      );
+    });
+  }
+
+  isDisposed: boolean;
+
+  dispose(): void {
+    if (this.isDisposed) {
+      return;
+    }
+    this.isDisposed = true;
+    Signal.clearData(this);
+  }
+
+  protected onCurrentFrameChanged(
+    callstackModel: Callstack.Model,
+    frame: Callstack.IFrame
+  ) {
+    const debugSessionPath = this.debuggerService.session.client.path;
+    this.handleNotebooks(debugSessionPath, frame);
+    this.handleConsoles(debugSessionPath, frame);
+    this.handleEditors(debugSessionPath, frame);
```

Comment:
These methods could be renamed `findInNotebooks`, `findInConsoles` and `findInEditors` (or something similar).

Reply:
Done.